src/Promitor.Integrations.LogAnalytics/LogAnalyticsClient.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Identity;
+using Azure.Monitor.Query;
+using Microsoft.Azure.Management.ResourceManager.Fluent;
+using Microsoft.Extensions.Logging;
+
+namespace Promitor.Integrations.LogAnalytics
+{
+    public class LogAnalyticsClient
+    {
+        private readonly ILogger _logger;
+        public readonly string ColumnNameResult = "result";
+        private readonly Uri _defaultEndpoint = new Uri("https://api.loganalytics.io");
+        private readonly Uri _govEndpoint = new Uri("https://api.loganalytics.us");
+
+        private readonly LogsQueryClient _logsQueryClient;
+
+        public LogAnalyticsClient(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<LogAnalyticsClient>();
+            _logsQueryClient = new LogsQueryClient(new DefaultAzureCredential());
+        }
+
+        public LogAnalyticsClient(ILoggerFactory loggerFactory, AzureEnvironment azureEnvironment, TokenCredential tokenCredentials)
+        {
+            _logger = loggerFactory.CreateLogger<LogAnalyticsClient>();
+            var uri = _defaultEndpoint;
+            if (azureEnvironment.Name == nameof(AzureEnvironment.AzureUSGovernment))
```

Comment:
What about other clouds?

Reply:
- With azureCN I found this https://learn.microsoft.com/en-us/azure/china/resources-developer-guide but there is nothing related to LogAnalytics
- With azureGermany I cannot find anything

Reply:
Normally the SDK provides an overview of these

Reply:
Too bad; it's OK to skip the rest as long as we open an issue to track this!

Reply:
And let's make sure we have a check to throw an exception if the cloud is not public nor US GOV so that end-users are aware of the limitation

Reply:
Ok. I will create an issue to track this after it is merged into master